glide-core/src/client/mod.rs

```diff
@@ -98,10 +98,67 @@ async fn run_with_timeout<T>(
     timeout: Duration,
     future: impl futures::Future<Output = RedisResult<T>> + Send,
 ) -> redis::RedisResult<T> {
-    tokio::time::timeout(timeout, future)
-        .await
-        .map_err(|_| io::Error::from(io::ErrorKind::TimedOut).into())
-        .and_then(|res| res)
+    if timeout == Duration::from_secs(0) {
```

Comment:
I think using Option\<Duration\> is more elegant since it will be evident from the function signature that Duration is optional and not implicit when timeout == 0 seconds

Reply:
agreed. will change